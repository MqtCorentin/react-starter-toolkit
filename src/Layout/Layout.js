import React from 'react';
import Header from 'Layout/Header';
import Footer from 'Layout/Footer';
import TitleBar from 'Layout/TitleBar';

export const renderTitle = ({ children, ...propsTitle }) => <TitleBar {...propsTitle}>{children}</TitleBar>;
export const renderHeader = propsHeader => <Header {...propsHeader} />;
export const renderFooter = () => <Footer />;

const Layout = (Component, props = {}) => <Component {...props} title={renderTitle} header={renderHeader} footer={renderFooter} />;
export default Layout;
