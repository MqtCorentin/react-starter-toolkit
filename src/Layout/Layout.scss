@import '@axa-fr/react-toolkit-core/dist/assets/scss/core.scss';

.af-main {
  min-height: calc(100vh - (calc(var(--fullheaderHeight) + var(--footerHeight))));
}
