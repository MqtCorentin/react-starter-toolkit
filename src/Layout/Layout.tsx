import { withClassDefault, withClassModifier, WithClassModifierOptions, compose, identity } from '@axa-fr/react-toolkit-core';
import Header from 'Layout/Header';
import Footer from 'Layout/Footer';
import TitleBar from 'Layout/TitleBar';
import Menu from 'Layout/Menu';
import { ComponentPropsWithoutRef, ReactNode } from 'react';

export type TLayout = WithClassModifierOptions & {
  children?: ReactNode;
  className?: string;
  fullScreen?: boolean;
  disabled?: {
    header?: boolean;
    menu?: boolean;
    title?: boolean;
    footer?: boolean;
  };
  propsHeader?: ComponentPropsWithoutRef<typeof Header>;
  propsMenu?: ComponentPropsWithoutRef<typeof Menu>;
  propsTitle?: ComponentPropsWithoutRef<typeof TitleBar>;
  propsFooter?: ComponentPropsWithoutRef<typeof Footer>;
};

const disabledDefault = {
  header: false,
  menu: false,
  title: false,
  footer: false,
};
const DEFAULT_CLASSNAME = 'af-main';

const Layout = ({ className, children, propsHeader, propsMenu, propsTitle, propsFooter, fullScreen, disabled = disabledDefault }: TLayout) => (
  <>
    {!disabled.header && <Header {...propsHeader} fullScreen={fullScreen} />}
    {!disabled.menu && <Menu {...propsMenu} fullScreen={fullScreen} />}
    {!disabled.title && <TitleBar {...propsTitle} fullScreen={fullScreen} />}
    <main className={className}>{fullScreen ? children : <section className="container">{children}</section>}</main>
    {!disabled.footer && <Footer {...propsFooter} fullScreen={fullScreen} />}
  </>
);

const enhance = compose(identity<TLayout>(), withClassDefault(DEFAULT_CLASSNAME), withClassModifier());

export default enhance(Layout);
