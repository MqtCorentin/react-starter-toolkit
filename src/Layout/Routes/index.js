export { default, WithAuth } from './Routes';
