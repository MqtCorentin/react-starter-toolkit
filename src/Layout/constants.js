import { ROUTE_URL_DASHBOARD as DASHBOARD } from 'Dashboard/constants';
import { ROUTE_URL_MEMBERS as MEMBERS } from 'Members/constants';
import { ROUTE_URL_SEARCHMEMBERS as SEARCHMEMBERS } from 'SearchMembers/constants';
import { ROUTE_URL_SLASH as SLASH } from 'SlashDesignSystem/constants';
import { ROUTE_URL_NOTFOUND as NOTFOUND } from 'NotFound/constants';
import { ROUTE_URL_UNAUTHORIZE as UNAUTHORIZE } from 'Unauthorize/constants';
import { ROUTE_URL_MODAL as MODAL } from 'Demos/Modal/constants';

const ROUTE_URL = {
  DASHBOARD,
  MEMBERS,
  SEARCHMEMBERS,
  SLASH,
  NOTFOUND,
  UNAUTHORIZE,
  MODAL,
};

export default ROUTE_URL;
