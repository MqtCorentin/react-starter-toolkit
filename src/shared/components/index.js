export { default as ModalCommonBody } from './ModalCommonBody';
export { default as ModalCommonHeader } from './ModalCommonHeader';
export { default as ModalCommonFooter } from './ModalCommonFooter';
