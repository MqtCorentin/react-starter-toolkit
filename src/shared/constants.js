import { string, number, shape, oneOfType, arrayOf } from 'prop-types';

export const MSG_REQUIRED = 'Le champ est obligatoire';

export const MSG_FORMAT = 'Format invalide';

export const MSG_ERROR_FORMAT = 'Format de fichier incorrect';

export const rulesRequired = {
  required: {
    message: MSG_REQUIRED,
  },
};

export const rulePattern = {
  pattern: {
    regex: /^[0-9a-zA-Z\s-_]{1,50}$/,
    message: MSG_FORMAT,
  },
};

export const rulesErrorFormat = {
  required: {
    message: MSG_ERROR_FORMAT,
  },
};

export const STATUS_HTTP = {
  SUCCESS: 200,
  CREATED: 201,
  BAD_REQUEST: 400,
  UNAUTHORIZE: 401,
  FORBIDDEN: 403,
  NOTFOUND: 404,
  SERVER_ERROR: 500,
};

export const STATUS_API = {
  SUCCESS: STATUS_HTTP.SUCCESS,
  CREATED: STATUS_HTTP.CREATED,
  WARNING: 40,
  ERROR: 50,
};

export const STATUS_HTTP_MESSAGES = {
  [STATUS_HTTP.SUCCESS]: 'Succès: La requête a réussi',
  [STATUS_HTTP.CREATED]: 'Succès: Création réussie',
  [STATUS_HTTP.BAD_REQUEST]: 'Erreur: La syntaxe de la requête est erronée',
  [STATUS_HTTP.UNAUTHORIZE]: 'Erreur: Une authentification est nécessaire pour accéder à la ressource',
  [STATUS_HTTP.FORBIDDEN]: 'Erreur: Accès non autorisé',
  [STATUS_HTTP.NOTFOUND]: 'Erreur: Elément non trouvé',
  [STATUS_HTTP.SERVER_ERROR]: 'Erreur: Problème technique ! Contacter votre support',
};

export const TIMEOUT = 20000;
export const TIMEOUT_ERROR_MESSAGE = 'Erreur: la requête a mis trop de temps';

export const MODIFIER_CLASS = {
  success: 'success',
  disabled: 'disabled',
};

export const COMMENT = {
  NAME: 'comment',
  LABEL: 'Commentaire',
  SUBLABEL: 'Facultatif',
  MAXLENGTH: '300',
};

export const JSON_CONTENT_HEADER = {
  'Content-Type': 'application/json',
};

export const STORYBOOK = 'https://axaguildev.github.io/react-toolkit/latest/storybook/';
export const DESIGN_SYSTEM = 'https://axaguildev.github.io/design-system/';

export const ADMIN = 'Admin';
export const READER = 'Reader';
export const WRITER = 'Writer';
export const USER = 'User';

export const PROFILS = [ADMIN, READER, WRITER, USER];

export const DEFAULT_OPTION_LABEL = '- Sélectionner -';
export const DEFAULT_OPTION = {
  value: '',
  label: DEFAULT_OPTION_LABEL,
};

/**
 * Commons proptypes
 */

export const ContextPropTypes = shape({
  Provider: shape({ defaultValue: string, currentValue: string, changedBits: number }),
  Consumer: shape({ defaultValue: string, currentValue: string, changedBits: number }),
});

export const SelectPropTypes = arrayOf(
  shape({
    value: oneOfType([string, number]),
    label: string,
  }),
);

export const AnomalyPropTypes = shape({
  code: number,
  detail: string,
  label: string,
});

export const IdPropTypes = oneOfType([string, number]);
