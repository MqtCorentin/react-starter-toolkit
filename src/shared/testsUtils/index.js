export { default as emptyFunction } from './emptyFunction';
export { default as WrapperStaticRouter } from './WrapperStaticRouter';
export { default as renderWithWrapperStaticRouter } from './renderWithWrapperStaticRouter';
