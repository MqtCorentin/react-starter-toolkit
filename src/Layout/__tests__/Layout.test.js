import { renderWithWrapperStaticRouter } from 'shared/testsUtils';
import Layout from '../Layout';

describe('Layout', () => {
  it.each`
    disabled
    ${{ header: true }}
    ${{ title: true }}
    ${{ footer: true }}
    ${{ menu: true }}
  `('Should render when disabled: $disabled', ({ disabled }) => {
    const { asFragment } = renderWithWrapperStaticRouter(<Layout disabled={disabled} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
