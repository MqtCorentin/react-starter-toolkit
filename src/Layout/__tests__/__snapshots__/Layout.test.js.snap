// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Should <CustomComponent /> contain layout props and customProps when call Layout with  1`] = `
<DocumentFragment>
  <div>
    Hello
  </div>
</DocumentFragment>
`;

exports[`Layout Should redirect to forbidden page when user connected is not aurhorized 1`] = `
<DocumentFragment>
  <p>
    Redirection forbidden
  </p>
</DocumentFragment>
`;
